### ISSUE TITLE
Stuck on "Loading codes" screen

### ISSUE THREAD

**Issue by poutros** [ahezard/NitroHax#7]
The cheats.xml is the same and worked on the previous version of nitrohax.

**Comment by ApacheThunder** [comment]
My cheats.xml still works with it. Sounds like yours is a bit too large. Trim a few games off it and try again.

**Comment by poutros** [comment]
I used the one provided on the gbatemp forum. https://gbatemp.net/threads/nitrohax-homebrew-action-replay-alternative-now-working-as-a-cia.438638/

I thought 3 mb was the limit.

**Comment by ApacheThunder** [comment]
No, the limit is somewhere less. I don't really know what the real limit is as I have not tested for that. I say 2.5mb is the safe limit. I had a 2.1MB one that works.

**Comment by poutros** [comment]
Mine is 2.1 mb too. I also have an alternate one that's ~900kb. All from that post.

**Comment by ApacheThunder** [comment]
I'll test the alt one in a minute but if it works for me, it's a bug I can not reproduce.

**Comment by poutros** [comment]
I have them inside the "NitroHax" folder on the root of my sd card as specified on the original nitrohax github repo.

**Comment by ApacheThunder** [comment]
I spotted a mistake in the 900kb one. There's an extra <game> line at the start that has no closing tag isn't attached to anything. Check that the larger one is correct too

**Comment by poutros** [comment]
http://www.w3schools.com/xml/xml_validator.asp

This site spotted no errors on either and I can't find that extra line.

**Comment by ApacheThunder** [comment]
Hmm strip all games except the ones you own and try again. I can't get it to load either. Sounds like the cheat data base loading code needs to be fixed. I'll talk to ahezard about that.

**Comment by ApacheThunder** [comment]
I have a very small XML with just 4 or 5 games in it and it loads fine.

**Comment by poutros** [comment]
The small one loads fine.

**Comment by ApacheThunder** [comment]
Yeah seems the parsing code is a bit unstable in TWL mode. I'll have ahezard look into it.

**Comment by sambailey54** [comment]
Same issue, let me know if I can help.

**Comment by ApacheThunder** [comment]
Ahezard is now aware of this. He's going to rewrite the XML loading code when he has time. 

For now just keep the XML below 900KB and it should work. If still don't work, stript out all games you don't own from the XML.

**Comment by ApacheThunder** [comment]
This issue has been corrected. There is a few more things I want to fix up before making a release build. But once that release build is out, this issue will be closed. Ahezard was able to load 9+MB cheat.xml with the committed changes. Though the downside with it now is if your game doesn't match a game in the XML, it won't show any codes at all so you can't manually chose a game in the cheat code list. That is the last thing that needs fixed before making a release build.

**Comment by ApacheThunder** [comment]
This issue is now resolved as of the latest release build.

**Comment by ApacheThunder** [comment]
The change has been merged. You are on your own on compiling/testing it. I do not have access to my 3DS right now so can't test it myself. I did however test the change on NTR_Launcher (specifically the DSi build of it) and didn't notice any issues.

I could make a DSi build of NitroHax and test this there but that will take some work as I have to remove some things (the touch screen mode switch is not needed on a HiyaCFW'ed DSi. :P ) among other things and that's not something I want to do right now. :P

Also the flicker issue Chishm mentioned may not be a problem on the 3DS build if you enable TWL clock speeds.
